spec/controllers/application_controller_spec.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe ApplicationController, type: :controller do
+  login
+  let(:user) { controller.current_user }
+
+  describe "#current_order" do
+    describe "current user present" do
+      it "returns the latest order of current user" do
+        FactoryBot.create(:order, user: user)
+        current_order = user.orders.last
+        expect(current_order).to be_present
+      end
+    end
```

Comment:
これでもまだ正しいことをテストできていないと思います。
ちょっと難しいかもしれないので、英語で書きますね。

`ApplicationController#current_order` returns `current_user.orders.last` if current_user exists and an order already exists for that user.

Therefore, what you really want to test is that when you call `controller.current_order`, the method returns the latest order for `current_user`.

What this rspec is currently doing
* Create an order for current_user using factory, and call that order `current_order`
* Check that `current_order` is present (which is obviously true because you just created it using the factory)

What this rspec should actually be doing
* Create an order for current_user using factory
* Call `controller.current_order` and see if the return value is equal to the order you just created via factory

Reply:
はい、分かりました。